From ddb2c8ae0baf3b73d77d1f6ce6142e92faa632af Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20K=C3=BChl?= <martin.kuehl@posteo.net>
Date: Fri, 30 Dec 2022 20:07:49 +0100
Subject: [PATCH] widget: Use correct end row for getting the selected text

The range is end-exclusive, so use end_row() instead of last_row().

Fixes: https://gitlab.gnome.org/GNOME/vte/-/issues/2584
---
 src/vtegtk.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/vtegtk.cc b/src/vtegtk.cc
index 3945ba67..591c127e 100644
--- a/src/vtegtk.cc
+++ b/src/vtegtk.cc
@@ -5699,7 +5699,7 @@ try
         return vte_terminal_get_text_range(terminal,
                                            selection.start_row(),
                                            selection.start_column(),
-                                           selection.last_row(),
+                                           selection.end_row(),
                                            selection.end_column(),
                                            nullptr,
                                            nullptr,
-- 
GitLab

